### DISCUSSION TITLE
part7 aws cloud watch 테라폼 실습 구성관련

### DISCUSSION THREAD

**Discussion by LSW4701** [tedilabs/fastcampus-devops#82] [Q&A] [answered]
박병진 강사님과는 EKS등등 테라폼 관련 실습을 완료했지만
part6 김동진 강사님과 강의를 떠나서 실습환경을 구성해 나가는것 자체에 많은 어려움이 있습니다. 

강사님께서 제공해주시는 테라폼 코드를 수강생 환경에 맞춰서 변경 해줘야되는데
해당 강사님은 실습환경 구현과정을 그대로 건너뛰니까

수강생 입장에서는 이게 오타가 나서 안되는건지, 강의자료 자체가 잘못된건지(part6, 젠킨스 파일 바뀐것 처럼) 
도무지 알수가 없습니다. 

하나하나 구글링 하면서 찾아가고 있지만 어떨때는 시간을 버리는것 같아 너무 화가나기도 합니다. 

**ㅁ 실습** 
- 김동진 강사님 git  https://github.com/dev-chulbuji/devops_infra.git

1. /apne2/prod/network/vpc 에서 terraform init 및 apply 완료
2. /apne2/prod/ec2/dmz_app 에서 terraform apply 과정에서 아래의 애러 발생 
error) 
module.iam.aws_iam_role.this[0]: Creating...
╷
│ Error: failed creating IAM Role (dmz-app-prod-role): AccessDenied: User: arn:aws:iam::893923049642:user/admin is not authorized to perform: iam:CreateRole on resource: arn:aws:iam::893923049642:role/dmz-app-prod-role with an explicit deny in an identity-based policy
│ 	status code: 403, request id: 77a9b2ad-97e0-46e9-9e66-449b940a1595
│ 
│   with module.iam.aws_iam_role.this[0],
│   on .terraform/modules/iam/modules/iam-assumable-role/main.tf line 75, in resource "aws_iam_role" "this":
│   75: resource "aws_iam_role" "this" {

![1111](https://user-images.githubusercontent.com/37770217/196058474-5afc8283-b28c-4c83-adc0-6b1bddfdc953.png)

구글링한 결과 현재는 주석처리된 apne2/prod/ec2/dmz_app/_backend.tf 의 role_arn 부분을 넣어줘야하는거같은데
어디서부터 어떻게 설정해야되는 모르겠습니다.
1) aws 콘솔에서 iam 신규 등록?  어떤정책을? 
2) https://registry.terraform.io/providers/hashicorp/aws/latest/docs 과 관련이 있는지?

**Comment by LSW4701** [comment, marked as answer]
해결완료했습니다.
아래의 글링크를 참고한뒤 iam access가 등록된 assume_role arn을 등록하면 됩니다.

- assume_role 이해   [https://brunch.co.kr/@topasvga/1201](https://brunch.co.kr/%40topasvga/1201)
-assume_role 생성 https://whchoi98.gitbook.io/aws-iam/iam-role

**Reply by posquit0** [reply]
에고.. 해결되셨다니 다행이네요~
제가 요즘 업무가 바빠 확인이 늦었습니다. 본 레포지토리는 패스트캠퍼스 DevOps초격차 중 박병진 강사 파트에 대해서만 질문응답을 받고 있습니다.
김동진 강사님에게 문의를 하기 위해서는 패스트캠퍼스를 통해 문의를 주셔야할 것 같습니다. 다음에 비슷한 이슈가 생기신다면 참고 부탁드립니다.

감사합니다.
